src/components/AccountDropdown.tsx

```diff
@@ -1,62 +1,96 @@
-// Since Account summary is using Address which still has the 'copy', we must handle hover ourself.
-
-interface IAccountDropdownProps {
+export interface IAccountDropdownProps {
   accounts: StoreAccount[];
   name: string;
-  value: StoreAccount;
+  value: StoreAccount | null;
   asset?: Asset;
+  clearable?: boolean;
   onSelect(option: StoreAccount): void;
 }
 
-const sortByLabel = (a: Option, b: Option) => a.label.localeCompare(b.label);
+// Combine Account and Asset into Option to simplify prop
+// passing. The form still expects `StoreAccount` as value,
+// so we make sure that we only return `option.account`
+export interface TAccountDropdownOption {
+  account: StoreAccount;
+  asset: {
+    balance: string;
+    assetSymbol: TSymbol;
+    assetUUID: TUuid;
+  };
+}
+
+const sortByLabel = (a: TAccountDropdownOption, b: TAccountDropdownOption) =>
+  a.account.label.localeCompare(b.account.label);
 
-type Option = StoreAccount & { balance: string; assetSymbol: string };
+const getOption = (account: StoreAccount, options: TAccountDropdownOption[]) =>
+  options.find((o) => o.account.uuid === account.uuid)!;
 
-function AccountDropdown({ accounts, name, value, onSelect, asset }: IAccountDropdownProps) {
-  const relevantAccounts: Option[] = accounts
-    .map((account) => ({
-      ...account,
-      balance: formatEther(asset ? getAccountBalance(account, asset) : getAccountBalance(account)),
-      assetUUID: asset ? asset.uuid : getBaseAsset(account)!.uuid,
-      assetSymbol: asset ? asset.ticker : getBaseAsset(account)!.ticker
+function AccountDropdown({
+  accounts,
+  asset,
+  name,
+  value: formValue,
+  onSelect
+}: IAccountDropdownProps) {
+  const options: TAccountDropdownOption[] = accounts
+    .map((a) => ({
+      account: a,
+      asset: {
+        balance: formatEther(asset ? getAccountBalance(a, asset) : getAccountBalance(a)),
+        assetUUID: asset ? asset.uuid : getBaseAsset(a)!.uuid,
+        assetSymbol: (asset ? asset.ticker : getBaseAsset(a)!.ticker) as TSymbol
+      }
     }))
     .sort(sortByLabel);
 
-  useEffectOnce(() => {
-    // preselect first value from options
-    if ((!value || !value.address) && relevantAccounts.length > 0) {
-      onSelect(relevantAccounts[0]);
-    }
-  });
+  const handleFormUpdate = (option: TAccountDropdownOption) => onSelect(option.account);
 
   return (
-    <Dropdown
+    <Selector<TAccountDropdownOption>
       name={name}
       placeholder={translateRaw('ACCOUNT_SELECTION_PLACEHOLDER')}
-      options={relevantAccounts}
-      onChange={(option) => onSelect(option)}
-      optionComponent={AccountOption}
-      value={value && value.address ? value : undefined} // Allow the value to be undefined at the start in order to display the placeholder
-      valueComponent={({ value: { uuid, label, address, balance, assetSymbol, assetUUID } }) => {
-        const option = relevantAccounts.find((a) => a.uuid === uuid);
+      searchable={true}
```

Comment:
The component isn't searchable since no `label` has been added to the `options`.

Reply:
Done with `getOptionLabel`